client/app/vps/cloud-database/order/vps-cloud-database-order.controller.js

```diff
@@ -0,0 +1,176 @@
+            ].join(" ")))
+            .finally(() => {
+                this.loading.durations = false;
+            });
+    }
+
+    cancelFurtherChoices () {
+        this.$timeout.cancel(this.redirectionTimeout);
+        this.loading.purchaseOrder = false;
+        this.loading.redirection = false;
+
+        this.currentOrder.duration = null;
+        this.currentOrder.contractsAccepted = false;
+        this.purchaseOrder = null;
+    }
+
+    getPricesForEachDuration (durations, version, ram, datacenter) {
+        return this.$q.all(_.map(
+            durations,
+            duration => this.getPrices(duration, version, ram, datacenter)
+        )).catch(error => this.CloudMessage.error([
+            this.$translate.instant("vps_tab_cloud_database_order_fetch_prices_failed"),
+            _(error).get("data.message", error)
+        ].join(" ")));
+    }
+
+    getPrices (duration, version, ram, datacenter) {
+        return this.ApiOrderDb.getNewDetails({ duration: duration.value, version, ram, datacenter }).$promise
+            .then(details => {
+                duration.details = details;
+            });
+    }
+
+    canOrder () {
+        return !_.any(this.loading) &&
+            this.currentOrder.version &&
+            this.currentOrder.ram &&
+            this.currentOrder.datacenter &&
+            this.currentOrder.duration &&
+            this.currentOrder.contractsAccepted;
+    }
+
+    generatePurchaseOrder () {
+        this.loading.purchaseOrder = true;
+        this.loading.redirection = true;
+        this.ApiOrderDb.orderNew(
+            { duration: this.currentOrder.duration.value },
+            {
+                version: this.currentOrder.version.value,
+                ram: this.currentOrder.ram.value,
+                datacenter: this.currentOrder.datacenter.value,
+                offer: "public"
+            }).$promise
+            .then(order => {
+                this.purchaseOrder = order;
+                this.redirectionTimeout = this.$timeout(() => {
+                    this.loading.redirection = false;
+                    this.openPurchaseOrder();
+                }, 5000);
+            })
+            .catch(error => this.CloudMessage.error([
+                this.$translate.instant("vps_tab_cloud_database_order_failed"),
+                _(error).get("data.message", error)
+            ].join(" ")))
+            .finally(() => {
+                this.loading.purchaseOrder = false;
+            });
+    }
+
+    openPurchaseOrder () {
+        this.atInternet.trackOrder({
+            name: `[sql-public]::${this.currentOrder.version.value}`,
+            page: "web::payment-pending",
+            orderId: this.purchaseOrder.orderId,
+            priceTaxFree: this.purchaseOrder.prices.withoutTax.value,
+            price: this.purchaseOrder.prices.withTax.value,
+            status: 1
+        });
+        this.$window.open(this.purchaseOrder.url);
```

Comment:
`_blank` and `noreferrer` ?

Reply:
I'm uncertain about whether we should keep this buggy feature of auto-opening the purchase order (the popup is blocked in modern browsers anyway) or just get rid of it.